reconcile/external_resources/manager.py

```diff
@@ -242,12 +249,22 @@ def _update_in_progress_state(
                 )
                 if r.action == Action.APPLY:
                     state.resource_status = ResourceStatus.PENDING_SECRET_SYNC
-                    state.reconciliation_errors = 0
                     self.state_mgr.set_external_resource_state(state)
                     need_secret_sync = True
                 elif r.action == Action.DESTROY:
                     state.resource_status = ResourceStatus.DELETED
                     self.state_mgr.del_external_resource_state(r.key)
+
+                metrics.set_gauge(
+                    ExternalResourcesReconcileTimeGauge(
+                        provision_provider=r.key.provision_provider,
+                        provisioner_name=r.key.provisioner_name,
+                        provider=r.key.provider,
+                        identifier=r.key.identifier,
+                        job_name=job_name,
+                    ),
+                    float(self.reconciler.get_resource_reconcile_duration(r) or 0),
```

Comment:
is convert to `float` required here? the actual result is `int`.

Reply:
nope. Fixed.